lib/slack/web/pagination/cursor.rb

```diff
@@ -7,23 +7,38 @@ class Cursor
 
           attr_reader :client
           attr_reader :verb
+          attr_reader :sleep_interval
+          attr_reader :max_retries
           attr_reader :params
 
           def initialize(client, verb, params = {})
             @client = client
             @verb = verb
+            @sleep_interval = params.delete(:sleep_interval)
+            @max_retries = params.delete(:max_retries) || client.default_max_retries
             @params = params
           end
 
           def each
             next_cursor = nil
+            retry_count = 0
             loop do
               query = { limit: client.default_page_size }.merge(params).merge(cursor: next_cursor)
-              response = client.send(verb, query)
+              begin
+                response = client.send(verb, query)
+              rescue Slack::Web::Api::Errors::TooManyRequestsError => e
+                raise e if retry_count >= max_retries
+                client.logger.debug("#{self.class}##{__method__}") { e.to_s }
+                retry_count += 1
+                sleep(e.retry_after.seconds)
+                next
+              end
               yield response
```

Comment:
You can move this inside the rescue block and avoid `next`, right?

Reply:
Also did I tell you that I was a nitpicky code reviewer? Hang on tight :)

Reply:
I understand, I'm the same way 😎.

Which part are you suggesting go in the rescue block? I started out with everything inside the begin block (see first commit b75bfcb) which avoids the `next`. Is this what you mean?

I changed it (8542d00) because I thought it was bad practice to have more lines than necessary inside the `begin` block — even if just for clarity (so that it's clear which line is expected to raise the error). But in this case it's obvious where the error is coming from, so I guess it's fine either way.

Reply:
Exactly.

Reply:
I actually now think yours is better. Someone can do something inside the yield that causes that exception and we're going to have bad side effects and catching an error we shouldn't.